src/hooks/useValidateRepository.ts

```diff
@@ -0,0 +1,188 @@
+
+import {dummySecret} from '@utils/sources';
+
+const fieldsNames: string[] = ['uri', 'token', 'username', 'branch', 'path'];
+
+const authSearchString = 'username or token:';
+const uriSearchString = 'uri:';
+const branchSearchString = 'branch:';
+const pathSearchString = 'path:';
+const endSearchString = ', context';
+
+const getErrorMessage = (rawErrorString: string, searchString: string) =>
+  rawErrorString.slice(
+    rawErrorString.search(searchString) + searchString.length,
+    rawErrorString.search(endSearchString)
+  );
+
+const useValidateRepository = (
+  getFieldValue: (name: NamePath) => string,
+  setValidationState: React.Dispatch<any>,
+  validateRepository: any
+) => {
+  const handleResponse = useCallback(
+    (res: any) => {
+      if (res?.error) {
+        const errorDetail = res?.error?.data?.detail;
+
+        if (!errorDetail) {
+          setValidationState({
+            message: 'Network error',
+            uri: 'error',
+          });
+          return;
+        }
+
+        if (errorDetail.includes(authSearchString)) {
+          setValidationState({
+            message: getErrorMessage(errorDetail, authSearchString),
+            uri: 'error',
+            token: getFieldValue('token') ? 'error' : '',
+            username: getFieldValue('username') ? 'error' : '',
+          });
+          return;
+        }
+        if (errorDetail.includes(uriSearchString)) {
+          setValidationState({
+            message: getErrorMessage(errorDetail, uriSearchString),
+            uri: 'error',
+          });
+          return;
+        }
+
+        if (errorDetail.includes(branchSearchString)) {
+          setValidationState({
+            message: getErrorMessage(errorDetail, branchSearchString),
+            uri: 'success',
+            token: getFieldValue('token') ? 'success' : '',
+            username: getFieldValue('username') ? 'success' : '',
+            branch: 'error',
+          });
+          return;
+        }
+
+        if (errorDetail.includes(pathSearchString)) {
+          setValidationState({
+            message: getErrorMessage(errorDetail, pathSearchString),
+            uri: 'success',
+            token: getFieldValue('token') ? 'success' : '',
+            username: getFieldValue('username') ? 'success' : '',
+            branch: getFieldValue('branch') ? 'success' : '',
+            path: 'error',
+          });
+          return;
+        }
+
+        setValidationState({
+          message: 'Unknown error',
+          uri: 'error',
+        });
+      } else {
+        setValidationState({
+          uri: 'success',
+          token: getFieldValue('token') ? 'success' : '',
+          username: getFieldValue('username') ? 'success' : '',
+          branch: getFieldValue('branch') ? 'success' : '',
+          path: getFieldValue('path') ? 'success' : '',
+        });
+      }
+    },
+    [setValidationState, getFieldValue]
+  );
+
+  const getValidationPayload = useCallback(() => {
```

Comment:
It doesn't have to be wrapped in `useCallback`, as it's not passed anywhere as dependency (no strict equal check)